src/services/ecommerce/payments/hooks.ts

```diff
@@ -0,0 +1,100 @@
+import { useQuery, useQueryClient, useMutation } from "@tanstack/react-query";
+import { paymentsApi } from "../client";
+import type {
+  PaymentsApiPaymentsBasketsListRequest,
+  PaymentsApiPaymentsBasketsCreateFromProductCreateRequest,
+  PaymentsApiPaymentsBasketsAddDiscountCreateRequest,
+  PaymentsApiPaymentsCheckoutCreateRequest,
+} from "../generated/v0/api";
+
+const usePaymentsBasketList = (
+  options: PaymentsApiPaymentsBasketsListRequest,
+) =>
+  useQuery({
+    queryKey: ["paymentsBaskets"],
+    queryFn: async () => {
+      const response = await paymentsApi.paymentsBasketsList(options);
+      return response.data;
+    },
+  });
+
+const useDeferredPaymentsBasketList = (
+  options: PaymentsApiPaymentsBasketsListRequest,
+  enabled: boolean,
+) =>
+  useQuery({
+    queryKey: ["paymentsBaskets"],
+    queryFn: async () => {
+      const response = await paymentsApi.paymentsBasketsList(options);
+      return response.data;
+    },
+    enabled: enabled,
+  });
+
+const usePaymentsBasketRetrieve = (id: number) => {
+  return useQuery({
+    queryKey: ["paymentsBaskets", id],
+    queryFn: async () => {
+      const response = await paymentsApi.paymentsBasketsRetrieve({ id });
+      return response.data;
+    },
+  });
+};
+
+const useDeferredPaymentsBasketRetrieve = (id: number, enabled: boolean) => {
+  return useQuery({
+    queryKey: ["paymentsBaskets", id],
+    queryFn: async () => {
+      const response = await paymentsApi.paymentsBasketsRetrieve({ id });
+      return response.data;
+    },
+    enabled: enabled,
+  });
+};
+
+const usePaymentsBaksetCreateFromProduct = () => {
+  const client = useQueryClient();
+  return useMutation({
+    mutationFn: (
+      slugAndSku: PaymentsApiPaymentsBasketsCreateFromProductCreateRequest,
+    ) =>
+      paymentsApi
+        .paymentsBasketsCreateFromProductCreate(slugAndSku)
+        .then((response) => response.data),
+    onSuccess: (_data) => {
+      client.invalidateQueries({ queryKey: ["paymentsBaskets"] });
```

Comment:
**Comment:** If you change the query key above to `["paymentsBaskets", opts]` then this invalidation is still correct. Query keys invalidation is hierarararraarichal and (by default) match partially, so invaliding `["a", { x: 1}]` invalidates 
- `["a", { x: 1}]`
- `["a", { x: 1}, "b"]`
- `["a", { x: 1, y: 2, z: 3}]`
-  but NOT `["a", { x: 100}]` or `["a"]`.